core/src/test/java/org/apache/iceberg/avro/RandomAvroData.java

```diff
@@ -51,6 +53,66 @@ public static List<Record> generate(Schema schema, int numRecords, long seed) {
     return records;
   }
 
+  public static List<StructLike> generateStructLike(Schema schema, int numRecords, long seed) {
```

Comment:
I don't think it makes sense for this utility to generate data for the internal object model. It's named `RandomAvroData` and generates the Avro object model. I would expect the utility to generate internal data to be in `RandomInternalData`.

Also, the `generateList` and `generateMap` methods can be either moved to `RandomUtil` or just duplicated.